### PR TITLE
Pcq 1986

### PR DESCRIPTION
**Pull request by priyankaVerma21**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/PCQ-1986

### Change description ###
Whitelisting URL

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: test/unit/components/testStringUtils.js [hunk 1]
```diff
@@ -3,15 +3,308 @@
...
+        expect(result).to.equal('https://platform.hmcts.net');
+    });
+
+    it('should not modify a URL that starts with https://localhost', () => {
+        const url = 'https://localhost:3000';
+        const result = prefixHttps(url);
+        expect(result).to.equal('https://localhost:3000');
+    });
+
+    it('should not modify a URL that starts with http://localhost', () => {
```
Comment (MartinYSpasov) [inline test/unit/components/testStringUtils.js:38]: Duplicated test

File: test/unit/components/testStringUtils.js [hunk 2]
```diff
@@ -3,15 +3,308 @@
...
-describe('prefixHttps()', () => {
 
-    it('should add HTTPS to the URL if missing', (done) => {
-        expect(prefixHttps('https://test.com')).to.equal('https://test.com');
-        expect(prefixHttps('http://test.com')).to.equal('http://test.com');
-        expect(prefixHttps('test.com')).to.equal('https://test.com');
-        expect(prefixHttps('    test.com')).to.equal('https://test.com');
-        expect(prefixHttps('    test      .        com')).to.equal('https://test.com');
-        done();
+describe('prefixHttps', () => {
```
Comment (MartinYSpasov) [inline test/unit/components/testStringUtils.js:7]: We have many positive scenarios not many negatives. What would be the outcome of www.myfakeSite/platform.hmcts.net for example?

**Review by MartinYSpasov — CHANGES_REQUESTED** [review]

File: test/unit/components/testStringUtils.js (hunk 2 above)
Comment (MartinYSpasov) [reply test/unit/components/testStringUtils.js:7]: So this scenario is not covered then, anyone could with a little coding knowledge can bypass these URLs checks

Comment (priyankaVerma21) [reply test/unit/components/testStringUtils.js:7]: This is now changed to check against hostname

**Review by MartinYSpasov — APPROVED** [review]

**Pull request merged**